### PR TITLE
Set `Cache-Control` headers on the redirects so they themselves can be cached

### PR DESCRIPTION
**Pull request by abernix**
In the event that the underlying function cannot be invoked (for example, an AWS us-east-1 outage or a Netlify rate-limiting event), we can let intermediary CDNs (like the Netlify CDN that sits in front of our functions, or even another CDN that we add in front of that CDN, in theory) serve "stale", yet valid content that has been generated recently.

In our case (and this functions case) the cached content is really just the `Location:` header that gets sent back as a HTTP response, but even caching that is valuable in an outage since GitHub release artifacts might still be up.  In the future, we can add other caching layers, including to GitHub Artifacts themselves (E.g., falling back to S3 or GCS), but this doesn't try to do that.

### CODE REVIEW

File: src/lib/download.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export async function downloadEvent(
         body: `You are being redirected to ${endpoint}`,
         headers: {
           Location: endpoint,
+          "Cache-Control": "max-age=60, stale-if-error=30, stale-while-revalidate=30",
```
Comment (abernix) [inline src/lib/download.ts:48]: Thoughts on these values [MDN `Cache-Control` Ref](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)?

Comment (EverlastingBugstopper) [reply src/lib/download.ts:48]: what do you think about a separate cache control returned per endpoint? for instance: versioned installer/tarball contents seem much more static/cacheable than any of the `latest` endpoints that have to change more frequently

Comment (EverlastingBugstopper) [reply src/lib/download.ts:48]: think this value is small enough as is to just merge but if we want to raise the values more down the line we should probably only do so for the static versioned endpoints

Comment (abernix) [reply src/lib/download.ts:48]: @EverlastingBugstopper I do wonder if `stale-if-error=18000` would be okay — even for `latest` — to guard against, say, a 60 minute Lambda outage.  There's ordinarily revalidation that will occur on a shorter interval which will establish the freshness, but if the origin is literally not available, maybe we're less likely to be worrying about a `latest` being stale?

Comment (EverlastingBugstopper) [reply src/lib/download.ts:48]: hmmm yeah that makes sense. lets go for the 18000 value and keep an eye on things

**Comment by EverlastingBugstopper** [comment]
can we add cache-control headers for the `/tar/` endpoints as well? that way if github goes down we also cache the binary. maybe gets a little wonky since all we can really cache is the redirect...

**Review by EverlastingBugstopper — APPROVED** [review]

**Pull request merged**